### PR TITLE
Respect muted attribute

### PR DESCRIPTION
**Pull request by mmcc**
The `muted` fix is 88634b0. The rest of the changes are switching the module over to dot notation.

My personal opinion is that we should just do this in every file we touch from now on. If you all disagree, I'll revert 0792def.

### CODE REVIEW

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 0792defed1b781c6443f56afc5c21bc7966e8658
Build details: https://travis-ci.org/pam/video.js/builds/72550503

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by mmcc** [comment]
@pam retry

**Comment by heff** [comment]
Volume zero isn't the same as muted. I'm not sure this is quite right. When you unmute it should go back to the previously set volume. The controls should be checking if the player is muted, not only rely on volume zero.

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 0792defed1b781c6443f56afc5c21bc7966e8658
Build details: https://travis-ci.org/pam/video.js/builds/72551311

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by gkatsev** [comment]
Changing all the things to use dotnotation is great. Though, it would be nice if you can leave a comment in the PR view where the meat of the changes are so it'll be easier to see it.

**Comment by mmcc** [comment]
@heff I don't think I follow. This is the muted attribute on initial play and we don't have any notion of default or persisted volume across players (anymore), so at the point we're setting volume to 0 the current volume is going to be 100% unless someone sets the volume on the `<video>` element before initializing the player.

I agree that this changes if we decide to reimplement default / persisted volume down the road, but as it stands now, I'm struggling to think of an edge case where this fails given the current setup.

@gkatsev Yeah, definitely. The original description message includes a link to the commit with the actual changes, but I don't think it's noticeable enough (everyone just immediately clicks "files" anyway). Notedd

**Comment by gkatsev** [comment]
Given that video begin with volume at 1, the `muted` attribute would mean that when we begin playback we're quiet (seems like volume is 0) but if we click the unmute button, the volume should be restored to 1. If we just set the volume to zero, we are then unable to restore the volume to 1 via the unmute button (without special handling).

**Comment by mmcc** [comment]
Ah wow, you're totally right. I was going based off the mute icon, but it also goes to that on volume 0, not just on mute. Clicking to unmute is totally busted with this approach.

You guys win. This time.

soon

**Comment by mmcc** [comment]
Well...definitely should have used `#muted()` to begin with...but relevant changes are now in commit 93320dc.

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 93320dc7b3f8d7e6ab49fc3a7fc50a7da85f27aa
Build details: https://travis-ci.org/pam/video.js/builds/72873934

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by mmcc** [comment]
@heff @gkatsev bump for thoughts on this approach: 93320dc

Also considered just going with something like `this.muted(!!options.muted)`, but this felt easier to follow. Otherwise I think this fixes the issue raised by Steve and everything works as expected, but let me know if I'm missing an edge case.

**Comment by heff** [comment]
Cool. Is this enough to fix the issue?

I think we need to get consistent with how we handle these attributes. One way or the other. eg controls

**Comment by mmcc** [comment]
Yes, it fixes the issue outlined in #2358, and I'm struggling to think of edge cases with this one. Your reticence has me second guessing. The logic required to get controls to work is much more of a rats nest, so in terms of complexity this doesn't need to come close.

That being said, I agree on the consistency note. I'll move this closer to controls.

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: d26cc5eb01c2f1d13ed5e7e042c3d48617f4d9da
Build details: https://travis-ci.org/pam/video.js/builds/73261672

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by heff** [comment]
I just assumed the volume controls would need to be updated to check the
muted attribute. Not t in a place where I can check that easily.

-- 
Sent from mobile

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: bed80ebfb9c6090d3041b19719211165d0206ad3
Build details: https://travis-ci.org/pam/video.js/builds/75008433

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by mmcc** [comment]
Something pulled into master is causing these tests to break. Looking into it.

**Comment by gkatsev** [comment]
The `tt is undefined` issue is because of https://github.com/videojs/video.js/issues/2454 which I'll fix tomorrow.

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 318b3cd44b18fa61339ce58748232603473c7d55
Build details: https://travis-ci.org/pam/video.js/builds/75141646

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by mmcc** [comment]
@pam retry

and please, for the love, work this time.

**Comment by mmcc** [comment]
@pam retry

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 318b3cd44b18fa61339ce58748232603473c7d55
Build details: https://travis-ci.org/pam/video.js/builds/75169983

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by gkatsev** [comment]
LGTM

File: src/js/player.js [hunk 1]
```diff
@@ -503,7 +506,7 @@ class Player extends Component {
       'autoplay': this.options_.autoplay,
       'preload': this.options_.preload,
       'loop': this.options_.loop,
-      'muted': this.options_.muted,
+      'muted': this.muted_,
```
Comment (heff) [inline src/js/player.js]: Let's change this back to options.muted and remove the muted_ setting above. In the future we need to handle properties like this better. Decide where the state should live and how to pass that between techs.

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 73ebb7b55d5df7195149ba83ce8bd432c84224dd
Build details: https://travis-ci.org/pam/video.js/builds/75305784

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Pull request closed without merging**